src/VisualStudio/Core/Def/Implementation/TodoComments/VisualStudioTodoCommentsService.cs

```diff
@@ -237,5 +243,10 @@ public async Task ReportTodoCommentDataAsync(DocumentId documentId, ImmutableArr
             await ReportTodoCommentDataAsync(
                 document.Id, converted.ToImmutable(), cancellationToken).ConfigureAwait(false);
         }
+
+        public void Dispose()
+        {
+            _lazyConnection?.Dispose();
+        }
```

Comment:
i consider disposal logic to be a very critical part of understanding the state/lifetime of an object.  because of this, i *Really* prefer that Dispose be placed next to the constructor (i.e. it pairs the create/destroy logic).